src/MultiThreadedCaches.jl

```diff
@@ -170,9 +152,14 @@ function Base.get!(func::Base.Callable, cache::MultiThreadedCache{K,V}, key) whe
         if value_or_miss !== CACHE_MISS
             return value_or_miss::V
         end
-        future = get!(cache.base_cache_futures, key) do
+        future_or_miss = get(cache.base_cache_futures, key, CACHE_MISS)
+        future::Channel{V} = if future_or_miss === CACHE_MISS
```

Comment:
is there a need for this convert? I think the type assert on the second branch should be sufficient
```suggestion
        future = if future_or_miss === CACHE_MISS
```

Reply:
i think you're probably right. i tend to be super aggressive about spraying these annotations when working on code that allocates....... i probably don't need to be. 😅 👍

Reply:
Done! :) thanks. and confirmed that the test still passes 😎